src/sagas/wallet.js

```diff
@@ -151,10 +153,29 @@ export function* startWallet(action) {
     dispatch = _dispatch;
   });
 
-  const networkSettings = yield select(getNetworkSettings);
+  // Network settings either from store or redux state
+  let networkSettings;
+  // Custom network settings are persisted in the app storage
+  const customNetwork = STORE.getItem(networkSettingsKeyMap.networkSettings);
+  if (customNetwork) {
+    networkSettings = customNetwork;
+    // On custom network settings one may use a different
+    // URL for the services from the ones registered by default
+    // for mainnet and testnet in the lib, and the wallet must
+    // behave consistently to the URLs set
+    config.setExplorerServiceBaseUrl(networkSettings.explorerServiceUrl);
+    config.setServerUrl(networkSettings.nodeUrl);
+    config.setTxMiningUrl(networkSettings.txMiningServiceUrl);
+
+    // If the wallet is initialized from quit state it must
+    // update the network settings on redux state
+    yield put(networkSettingsUpdateState(networkSettings));
+  } else {
+    networkSettings = yield select(getNetworkSettings);
+  }
 
   let wallet;
-  if (useWalletService) {
+  if (useWalletService && !isEmpty(networkSettings.walletServiceUrl)) {
```

Comment:
I don't understand this... What if he never used the network settings but is using the wallet-service (which is now 100% of our userbase)? He will have an empty networkSettings value, right? Is this a bug?

Reply:
No. The `walletServiceUrl` has a value in redux state by default. Beyond that, the pre-settings for network settings, `mainnet` and `testnet`, also have default values for `walletServiceUrl`. So, wallet-service will work fine. It will not work only if the user customize the network settings removing the `walletServiceUrl`.

Reply:
Oh, I see!

I was confusing `devSettings` with `networkSettings`.

So at this point, `networkSettings` will always have a value, either populated from the `PRE_SETTINGS` or from the `devSettings`